src/core/shape_inference/include/slice_scatter_shape_inference.hpp

```diff
@@ -0,0 +1,140 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <array>
+
+#include "openvino/core/validation_util.hpp"
+#include "openvino/op/slice_scatter.hpp"
+#include "slice_shape_inference.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v15 {
+template <class T, class TRShape = result_shape_t<T>>
+std::vector<TRShape> shape_infer(const SliceScatter* op,
+                                 const std::vector<T>& input_shapes,
+                                 const ITensorAccessor& ta = make_tensor_accessor()) {
+    using DimType = typename T::value_type;
+    const auto& num_of_inputs = input_shapes.size();
+
+    NODE_SHAPE_INFER_CHECK(op,
+                           input_shapes,
+                           num_of_inputs == 5 || num_of_inputs == 6,
+                           "SliceScatter has to have 5 or 6 inputs. Got: ",
+                           num_of_inputs);
+    const auto input_shape = input_shapes[0];
+    const auto& data_rank = input_shape.rank();
```

Comment:
```suggestion
    const auto& input_shape = input_shapes[0];
    const auto& data_rank = input_shape.rank();
```
or (as `input_shape` is not needed more than once)
```suggestion
    const auto& data_rank = input_shapes[0].rank();
```